### PR TITLE
cleaned up TODOs

### PR DESCRIPTION
**Pull request by hythloda**
Fixes #?

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [x] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [x] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [x] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

**Comment by hythloda** [comment]
Only 1 TODO left regarding SPI. This will make it into 7.6 RC1 but is not ready yet.

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -269,10 +264,13 @@ Related issues:
 
 [Support "Zulu OpenJDK Discovery API" for auto provisioning toolchains gradle#19140](https://github.com/gradle/gradle/issues/19140)
 
-#### TODO: Enhanced the plugin declaration DSL from java-gradle-plugin
+#### Enhanced the plugin declaration DSL for the java-gradle-plugin
+
+The recently released [version 1.0.0 of the Plugin-Publish plugin](https://plugins.gradle.org/plugin/com.gradle.plugin-publish/1.0.0) significantly improves the process by having stronger opinions and a more straightforward configuration.
```
Comment (nathan-contino) [inline subprojects/docs/src/docs/release/notes.md:269]: I'm unclear what "the process" means here. Plugin declaration? Plugin publishing? Both?

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -269,10 +264,13 @@ Related issues:
 
 [Support "Zulu OpenJDK Discovery API" for auto provisioning toolchains gradle#19140](https://github.com/gradle/gradle/issues/19140)
 
-#### TODO: Enhanced the plugin declaration DSL from java-gradle-plugin
+#### Enhanced the plugin declaration DSL for the java-gradle-plugin
+
+The recently released [version 1.0.0 of the Plugin-Publish plugin](https://plugins.gradle.org/plugin/com.gradle.plugin-publish/1.0.0) significantly improves the process by having stronger opinions and a more straightforward configuration. 
+
+We enhance the `java-gradle-plugin` documentation, describing the Maven publications it generates when used together with the `maven-publish plugin`.
```
Comment (nathan-contino) [inline subprojects/docs/src/docs/release/notes.md:271]:
```suggestion
This release includes updates to the `java-gradle-plugin` documentation.
The enhanced documentation describes the Maven publications generated when
combined with the `maven-publish plugin`.
```

**Review by nathan-contino — APPROVED** [review]
LGTM -- a question and a suggestion.

File: subprojects/docs/src/docs/release/notes.md [hunk 3]
```diff
@@ -266,7 +266,7 @@ Related issues:
 
 #### Enhanced the plugin declaration DSL for the java-gradle-plugin
 
-The recently released [version 1.0.0 of the Plugin-Publish plugin](https://plugins.gradle.org/plugin/com.gradle.plugin-publish/1.0.0) significantly improves the process by having stronger opinions and a more straightforward configuration. 
+The recently released [version 1.0.0 of the Plugin-Publish plugin](https://plugins.gradle.org/plugin/com.gradle.plugin-publish/1.0.0) significantly improves the plugin publishing process by having stronger opinions and a more straightforward configuration.
```
Comment (nathan-contino) [inline subprojects/docs/src/docs/release/notes.md:269]:
```suggestion
The recently released [version 1.0.0 of the Plugin-Publish plugin](https://plugins.gradle.org/plugin/com.gradle.plugin-publish/1.0.0) significantly improves the plugin publishing process through stronger opinions and a more straightforward configuration.
```
Small wording improvement.

**Review by nathan-contino — APPROVED** [review]
LGTM!

**Comment by hythloda** [comment]
@bot-gradle test and merge

**Pull request merged**